src/robusta/core/sinks/robusta/rrm/prometheus_alert_resource_manager.py

```diff
@@ -0,0 +1,288 @@
+
+                sliced_alerts = sorted_active_rules[start_index:end_index]
+                _, item_length = self.__create_crd_file(name=name, alerts=sliced_alerts)
+                synced_item_count += item_length
+
+            # When a user disables or deletes alerts from the sass platform, they may be
+            # removed from [active_alert_rules] and consequently from the cluster
+            # In certain edge cases, there's a possibility that these rules aren't entirely
+            # cleared from the cluster or cache.
+            # This typically happens when the [max_iterations] value is smaller than the size of the cached dictionary.
+            # As a result, the above loop might not cover all crd file buckets, leaving
+            # behind residual rules in both the cluster and cache.
+            # The loop below ensures these residues are effectively cleaned up.
+            cached_length = len(self.__crd_files_cache.keys())
+            for next_iteration in range(max_iterations + 1, cached_length + 1):
+                name = f"{self.__crd_name}--{next_iteration}"
+                self.__delete_crd_file(name=name)
+
+            return synced_item_count
+        except Exception as e:
+            logging.error(f"Error occured while creating prometheus alerts: {e}", exc_info=True)
+
+            raise e
+
+    def __get_snapshot_body(self, name: str, rules: List[dict]):
+        return {
+            "apiVersion": self.__k8_apiVersion,
+            "kind": self.__kind,
+            "metadata": {
+                "name": name,
+                "labels": {
+                    "release": "robusta",
+                    "role": "alert-rules",
+                    self.__label_selector_key: self.__label_selector_value
+                },
+            },
+
+            "spec": {
+                "groups": [
+                    {
+                        "name": self.__group_name,
+                        "rules": rules
+                    }
+                ]
+            }
+        }
+
+    def __get_flattened_crd_files_cache_map(self) -> Dict[str, PrometheusAlertRule]:
+        flattened_map = {}
+
+        for sublist in self.__crd_files_cache.values():
+            for item in sublist:
+                # we need the deep copy to make sure than we do not modify the original cached object references
+                flattened_map[item.labels.entity_id] = copy.deepcopy(item)
+
+        return flattened_map
+
+    def __list_alerts(self):
+        try:
+            return self.__k8_api.list_namespaced_custom_object(
+                group=self.__group,
+                version=self.__version,
+                plural=self.__plural,
+                label_selector=self.__label_selector,
+                namespace=self.__installation_namespace
+            )
+        except Exception as e:
+            raise f"An error occured while listing PrometheusRules CRD: {e}"
+
+    def __custom_object_exists(self, name: str):
```

Comment:
What is the return value here?

Also, the name suggest this method returns a boolean, bug it returns the actual custom resource

shouldn't it be named `read_custom_resource` ?